src/io/iter_image_recordio_2.cc

```diff
@@ -134,19 +134,25 @@ inline void ImageRecordIOParser2<DType>::Init(
   record_param_.InitAllowUnknown(kwargs);
   batch_param_.InitAllowUnknown(kwargs);
   normalize_param_.InitAllowUnknown(kwargs);
+  PrefetcherParam prefetch_param;
+  prefetch_param.InitAllowUnknown(kwargs);
   n_parsed_ = 0;
   overflow = false;
   rnd_.seed(kRandMagic + record_param_.seed);
   int maxthread, threadget;
-  #pragma omp parallel
-  {
-    // be conservative, set number of real cores
-    maxthread = std::max(omp_get_num_procs(), 1);
-  }
-  param_.preprocess_threads = std::min(maxthread, param_.preprocess_threads);
-  #pragma omp parallel num_threads(param_.preprocess_threads)
-  {
-    threadget = omp_get_num_threads();
+  if (prefetch_param.ctx == PrefetcherParam::CtxType::kCPU) {
+    threadget = engine::OpenMP::Get()->GetRecommendedOMPThreadCount();
```

Comment:
Since prefetching is happening in parallel with op execution. won't this cause too many threads to be launched when we are also useing openmp threads to execute operators.

Reply:
No. For CPU optimized data loader, prefetching isn't happened.
Prefetching will flush cache, causing CPU op execution much slower. So for CPU optimized data loader, the loader itself works as a normal operator, which will be executed exclusively, utilizing all omp threads.

Reply:
ohh okay, I didnt realize that the code paths are different for CPU and GPU iterators now. LGTM :)